### PR TITLE
Refactor exit probability

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: consensus/exit_probability_randomizer/walker.c [hunk 1]
```diff
@@ -112,8 +108,8 @@ static retcode_t random_walker_select_approver_tail(
       *has_approver_tail = false;
       return ret;
     }
-    HASH_FIND(hh, approvers_entry->approvers, approver->trits,
-              FLEX_TRIT_SIZE_243, approver_entry);
+    HASH_FIND(hh, approvers_entry->approvers, approver, FLEX_TRIT_SIZE_243,
+              approver_entry);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/walker.c]: Maybe replace this `HASH_FIND` by a function as well ?

Comment (tsvisabo) [reply consensus/exit_probability_randomizer/walker.c]: sure

File: utils/hash_maps.c [hunk 2]
```diff
@@ -49,6 +49,20 @@ void hash_int_map_free(hash_int_map_t *const map) {
   *map = NULL;
 }
 
+bool hash_int_map_find(hash_int_map_t const *const map,
+                       flex_trit_t const *const hash,
+                       hash_to_int_map_entry_t const **res) {
+  if (*map == NULL) {
```
Comment (thibault-martinez) [inline utils/hash_maps.c]: Also check for map before dereferencing it ?

File: utils/hash_maps.c [hunk 3]
```diff
@@ -66,6 +80,20 @@ bool hash_to_indexed_hash_set_map_contains(
   return entry != NULL;
 }
 
+bool hash_to_indexed_hash_set_map_find(
+    hash_to_indexed_hash_set_map_t const *const map,
+    flex_trit_t const *const hash,
+    hash_to_indexed_hash_set_entry_t const **res) {
+  if (!(*map)) {
```
Comment (thibault-martinez) [inline utils/hash_maps.c]: Also check for map before dereferencing it ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
You forgot 3 model casts:
- https://github.com/iotaledger/entangled/blob/develop/consensus/milestone_tracker/milestone_tracker.c#L385
- https://github.com/iotaledger/entangled/blob/develop/gossip/node.c#L75
- https://github.com/iotaledger/entangled/blob/develop/gossip/node.c#L117

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**